### DISCUSSION TITLE
Can't access Prometheus and grafana pages using VM mapped ports after docker runs the demo

### DISCUSSION THREAD

**Discussion by just-a-learning-person** [open-telemetry/opentelemetry-demo#1434] [Q&A] [answered]
![image](https://github.com/open-telemetry/opentelemetry-demo/assets/55468780/7d2f1308-28c5-4833-aa8b-a64fe6f0bfcb)
I started the demo using docker, and the "docker ps" showed the result as shown in the figure. I access port 0.0.0.0:49161 on the front end page and the program can display the shopping page. I went on to access Prometheus port 0.0.0.0:9090 and it can bring up the Prometheus page, but there is no data, even if I click on the shopping page.

**Comment by julianocosta89** [comment]
Hey @just-a-learning-person the frontend app should be available at `http://localhost:8080/` and you can access the metrics on grafana, available at `http://localhost:8080/grafana`.

Let us know if you can see the metrics in grafana

**Reply by just-a-learning-person** [reply]
![image](https://github.com/open-telemetry/opentelemetry-demo/assets/55468780/f3a523d6-0dc6-4031-8ff2-c0127f2f9b3f)
![image](https://github.com/open-telemetry/opentelemetry-demo/assets/55468780/d0e50e7b-e7d5-4403-87f1-9eb6f7a0da57)
![image](https://github.com/open-telemetry/opentelemetry-demo/assets/55468780/2148be74-5f5a-4805-9fca-71fe8a8ed52a)

nope, can't do it.  am i suppose to install grafana on my vm? Centos7 btw.

**Comment by julianocosta89** [comment]
@just-a-learning-person looks like your `frontend-proxy` container is not running, as well as a couple of other services.

Could you share the logs from the `frontend-proxy` container ?

**Reply by just-a-learning-person** [reply]
[frontend-proxy.log](https://github.com/open-telemetry/opentelemetry-demo/files/14532665/frontend-proxy.log)
[other-container-report.md](https://github.com/open-telemetry/opentelemetry-demo/files/14532667/other-container-report.md)
Thank you for your patience. Here are the logs of 6 not-running pods.
Looks like this demo couldn't show the results right away once it was started. It still needs more dependencies and configuration.

**Comment by julianocosta89** [comment]
do you mind trying cleaning up everything and starting again?
Lets try this:

```
docker system prune -a

docker compose build

docker compose up
```

This will take some time depending on your machine and internet connection, but it would help investigate this further.
The errors you are getting are a bit odd for docker containers.
Like for java `ERROR: JAVA_HOME is set to an invalid directory: /opt/java/openjdk`.

You mentioned where your `JAVA_HOME` is set, but that doesn't matter at all, the `JAVA_HOME` here should be the one used within the container.

**Reply by just-a-learning-person** [reply]
`docker-compose build` got error: 
> ERROR [quoteservice base 3/4] RUN chmod +x /usr/local/bin/install-php-extensions   && install-php-extensions     opcache     pcntl     protobuf     opentelemetry 
failed to solve: executor failed running [/bin/sh -c chmod +x /usr/local/bin/install-php-extensions   && install-php-extensions     opcache     pcntl     protobuf     opentelemetry]: exit code: 1
Hope this would help.

**Comment by julianocosta89** [comment]
@just-a-learning-person could you try again without the `-`.
`docker-compose` is `v1`  and it was deprecated in favour of `docker compose`, which is now `v2`.
yeah, I know, naming is great... hahaha

**Reply by just-a-learning-person** [reply]
Sorry, I used the standalone version of docker compose, which is `docker-compose`. If I type in `docker compose`, it would be `docker: 'compose' is not a docker command.`. Reference: [standalone version of docker-compose](https://docs.docker.com/compose/install/standalone/)

**Reply by julianocosta89** [reply]
which docker version are you using?

**Reply by just-a-learning-person** [reply]
Docker version 20.10.3, build 48d30b5
Is there a restriction on docker version in the [documentation](https://opentelemetry.io/docs/demo/docker-deployment/)?

**Reply by julianocosta89** [reply, marked as answer]
We never thought so, but now, I'm wondering that maybe yes.
We do require docker compose v2, as `v1` was deprecated: https://docs.docker.com/compose/reference/

Could you try updating your docker version or install the compose v2?
https://docs.docker.com/compose/install/linux/#install-using-the-repository

**Reply by julianocosta89** [reply]
@just-a-learning-person did you have the chance to try that out?
Is this issue solved?

**Reply by just-a-learning-person** [reply]
Yes, I did. Fortunately it did work. 
Thanks for your help.
